PowerEditor/src/ScintillaComponent/UserDefineDialog.cpp

```diff
@@ -925,9 +925,19 @@ void UserDefineDialog::reloadLangCombo()
 void UserDefineDialog::changeStyle()
 {
     _status = !_status;
-    ::SetDlgItemText(_hSelf, IDC_DOCK_BUTTON, (_status == DOCK)?TEXT("Undock"):TEXT("Dock"));
+    NativeLangSpeaker *pNativeSpeaker = (NppParameters::getInstance()).getNativeLangSpeaker();
+    TiXmlNodeA *targetNode = nullptr;
+    if (pNativeSpeaker->getNativeLangA())
+    {
+        targetNode = (pNativeSpeaker->getNativeLangA())->FirstChildElement("Dialog");
+        if (targetNode)
+            targetNode = targetNode->FirstChildElement("UserDefine");
+    }
+    generic_string dockStr = pNativeSpeaker->getAttrNameByIdStr(TEXT("Dock"), targetNode, std::to_string(IDC_DOCK_BUTTON).c_str());
+    generic_string undockStr = pNativeSpeaker->getAttrNameByIdStr(TEXT("Undock"), targetNode, std::to_string(IDC_UNDOCK_BUTTON).c_str());
+    ::SetDlgItemText(_hSelf, IDC_DOCK_BUTTON, (_status == DOCK) ? undockStr.c_str() : dockStr.c_str());

```

Comment:
I would do somethings similar as following code:
```cpp
generic_string dockButtonLabel;
generic_string defauleLabel;
generic_string idStr;
if (_status == DOCK)
{
    defauleLabel = TEXT("Undock");
    idStr = std::to_string(IDC_UNDOCK_BUTTON);
}
else
{
    defauleLabel = TEXT("Dock");
    idStr = std::to_string(IDC_DOCK_BUTTON);
}

dockButtonLabel= pNativeSpeaker->getAttrNameByIdStr(defauleLabel.c_str(), targetNode, idStr.c_str());
::SetDlgItemText(_hSelf, IDC_DOCK_BUTTON, dockButtonLabel.c_str());
```

So we don't call `getAttrNameByIdStr` twice.